packages/babel-helpers/src/helpers.js

```diff
@@ -588,28 +588,43 @@ helpers.interopRequireDefault = helper("7.0.0-beta.0")`
 `;
 
 helpers.interopRequireWildcard = helper("7.0.0-beta.0")`
+  function _getRequireWildcardCache() {
+    if (typeof WeakMap !== "function") return null;
+
+    var cache = new WeakMap();
+    _getRequireWildcardCache = function () { return cache; };
+    return cache;
+  }
+
   export default function _interopRequireWildcard(obj) {
     if (obj && obj.__esModule) {
       return obj;
-    } else {
-      var newObj = {};
-      if (obj != null) {
-        for (var key in obj) {
-          if (Object.prototype.hasOwnProperty.call(obj, key)) {
-            var desc = Object.defineProperty && Object.getOwnPropertyDescriptor
-              ? Object.getOwnPropertyDescriptor(obj, key)
-              : {};
-            if (desc.get || desc.set) {
-              Object.defineProperty(newObj, key, desc);
-            } else {
-              newObj[key] = obj[key];
-            }
+    }
+
+    var cache = _getRequireWildcardCache();
+    if (cache && cache.has(obj)) {
+      return cache.get(obj);
+    }
+
+    var newObj = {};
+    if (obj != null) {
+      var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
+      for (var key in obj) {
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
+          var desc = hasPropertyDescriptor
+            ? Object.getOwnPropertyDescriptor(obj, key)
+            : null;
+          if (desc && (desc.get || desc.set)) {
+            Object.defineProperty(newObj, key, desc);
+          } else {
+            newObj[key] = obj[key];
           }
         }
       }
-      newObj.default = obj;
-      return newObj;
     }
+    newObj.default = obj;
+    cache && cache.set(obj, newObj);
```

Comment:
Nit: Use an if statement.

Reply:
@jridgewell Thanks for your advice. I modified it.